In [1]:
# Imports
import numpy as np
import pandas as pd
import scipy.stats as stats

In [2]:
# Load the data
df = pd.read_csv('data/weight-height.csv')
df.head()

Gender     Height      Weight
0   Male  73.847017  241.893563
1   Male  68.781904  162.310473
2   Male  74.110105  212.740856
3   Male  71.730978  220.042470
4   Male  69.881796  206.349801

In [7]:
data = df[df['Gender'] == 'Female']['Height']
data

5000    58.910732
5001    65.230013
5002    63.369004
5003    64.479997
5004    61.793096
          ...    
9995    66.172652
9996    67.067155
9997    63.867992
9998    69.034243
9999    61.944246
Name: Height, Length: 5000, dtype: float64

In [9]:
mean = np.mean(data)
mean

63.708773603458006

In [10]:
std = np.std(data)
std

2.696014373906208

In [11]:
stats.norm.cdf(56, loc = mean, scale = std) -\
stats.norm.cdf(55, loc = mean, scale = std)

0.0015043054289559745

> The probability of a woman being between 55 inches and 56 inches in height is about 0.15%

In [12]:
stats.norm.cdf(60, loc = mean, scale = std)

0.0844649272520746

> The probability of a woman being less than 5 feet tall is about 8.4%

> Since this data is approximately normal, the probability of a woman being taller than the mean height is about 50%.